src/Accounts/Authentication/Factories/AuthenticationFactory.cs

```diff
@@ -475,7 +475,7 @@ private string GetEndpointToken(IAzureAccount account, string targetEndpoint)
             {
                 tokenKey = AzureAccount.Property.KeyVaultAccessToken;
             }
-            if (string.Equals(targetEndpoint, AzureEnvironment.ExtendedEndpoint.MicrosoftGraphEndpointResourceId, StringComparison.OrdinalIgnoreCase))
+            if (string.Equals(targetEndpoint, AzureEnvironment.ExtendedEndpoint.MicrosoftGraphEndpointResourceId, StringComparison.OrdinalIgnoreCase) || string.Equals(targetEndpoint, AzureEnvironment.ExtendedEndpoint.MicrosoftGraphUrl, StringComparison.OrdinalIgnoreCase))
```

Comment:
Should we do this for key vault as well?

Reply:
> Should we do this for key vault as well?

if you mean allow "AzureKeyVaultDnsSuffix" to get keyvault accesstoken I think it's safe to add this key.
But considering this key was passed in by ClientFactory.CreateArmClient, it's less likely the suffix will be used to create service client. Do you have other use case of this method which suffix could be passed in?

Reply:
In that case I suggest adding a comment reminding others not to follow the practice of MSGraph.

Reply:
> In that case I suggest adding a comment reminding others not to follow the practice of MSGraph.

create issue: https://github.com/Azure/azure-powershell/issues/18634 to describe the ideal fix